docs/reference-guides/block-api/block-deprecation.md

```diff
@@ -38,7 +38,7 @@ Deprecations are defined on a block type as its `deprecated` property, an array
 -   `supports` (Object): The [supports definition](/docs/reference-guides/block-api/block-registration.md) of the deprecated form of the block.
 -   `save` (Function): The [save implementation](/docs/reference-guides/block-api/block-edit-save.md) of the deprecated form of the block.
 -   `migrate` (Function, Optional): A function which, given the old attributes and inner blocks is expected to return either the new attributes or a tuple array of `[ attributes, innerBlocks ]` compatible with the block. As mentioned above, a deprecation's `migrate` will not be run if its `save` function does not return a valid block so you will need to make sure your migrations are available in all the deprecations where they are relevant.
--   `isEligible` (Function, Optional): A function which, given the attributes and inner blocks of the parsed block, returns true if the deprecation can handle the block migration even if the block is valid. This is particularly useful in cases where a block is technically valid even once deprecated, but still requires updates to its attributes or inner blocks. This function is not called when the results of all previous deprecations' `save` functions were invalid.
+-   `isEligible` (Function, Optional): A function which, given the attributes and inner blocks of the parsed block, returns true if the deprecation can handle the block migration even if the block is valid. This is particularly useful in cases where a block is technically valid even once deprecated, but still requires updates to its attributes or inner blocks. If further information is required to determine this deprecation's eligibility, the block node and object are provided within a third parameter. This function is not called when the results of all previous deprecations' `save` functions were invalid.
```

Comment:
What do you think about re-structuring the text for `isEligible` (and maybe `migrate` too)?

A few changes that I think would be good:
- Document the arguments using bullet points (matching what we do for function that have JSDoc types).
- Detail the expected return type (again like we do for JSDoc types).
- Ensure the terms we're using are consistent (e.g. parsed block and block node are both mentioned currently).
- Add some more clarity to that last sentence. I've read it a few times and still don't understand it.

Ideally `isEligible` and `migrate` are so complicated as concepts that they should be in a separate section, but that's not one for this PR.

Reply:
Some great suggestions there, thank you 👍 

I've tried reformatting and rewording the `isEligible` and `migrate` function docs in https://github.com/WordPress/gutenberg/pull/48815/commits/efd2cd7bd6f484ada942433ed4fc32dc4786e261. 

Let me know if any further tweaks are needed in this PR.